iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/AliasesTest.java

```diff
@@ -0,0 +1,92 @@
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.test.NetworkTest1Factory;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Sebastien Murgey <sebastien.murgey at rte-france.com>
+ */
+public class AliasesTest {
+    @Test
+    public void canAddAliases() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        assertTrue(load.getAliases().isEmpty());
+        load.addAlias("Load alias");
+
+        assertEquals(1, load.getAliases().size());
+        assertTrue(load.getAliases().contains("Load alias"));
+        assertNotNull(network.getLoad("Load alias"));
+        assertEquals(network.getLoad("Load alias"), load);
+    }
+
+    @Test
+    public void canRemoveAliases() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        assertTrue(load.getAliases().isEmpty());
+        load.addAlias("Load alias");
+
+        assertEquals(1, load.getAliases().size());
+        assertTrue(load.getAliases().contains("Load alias"));
+
+        load.removeAlias("Load alias");
+        assertTrue(load.getAliases().isEmpty());
+    }
+
+    @Test
+    public void silentlyIgnoreAffectingObjectsIdAsAlias() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        assertTrue(load.getAliases().isEmpty());
+        load.addAlias("load1");
+        assertTrue(load.getAliases().isEmpty());
+    }
+
+    @Test
+    public void silentlyIgnoreAffectingTwiceSameIdToAnObject() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        assertTrue(load.getAliases().isEmpty());
+        load.addAlias("Load alias");
+        load.addAlias("Load alias");
+
+        assertEquals(1, load.getAliases().size());
+        assertTrue(load.getAliases().contains("Load alias"));
+    }
+
+    @Test(expected = PowsyblException.class)
+    public void failWhenDuplicatedAlias() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        Generator generator = network.getGenerator("generator1");
+        load.addAlias("Alias");
+        generator.addAlias("Alias");
+    }
+
+    @Test(expected = PowsyblException.class)
+    public void failWhenAliasEqualToAnId() {
+        Network network = NetworkTest1Factory.create();
+        Generator generator = network.getGenerator("generator1");
+        generator.addAlias("load1");
+    }
+
+    @Test(expected = PowsyblException.class)
+    public void failWhenRemovingNonExistingAlias() {
+        Network network = NetworkTest1Factory.create();
+        Load load = network.getLoad("load1");
+        load.removeAlias("Load alias");
+    }
+}
```

Comment:
Could you please add a test where you try to add as an alias the ID of an identifiable, and ensure you can get the object.

Reply:
The ID of the identifiable you are adding an alias to, or of another identifiable ?